circus-train-gcp/src/main/java/com/hotels/bdp/circustrain/gcp/GCPCredentialCopier.java

```diff
@@ -15,77 +15,107 @@
 import java.net.URISyntaxException;
+import java.nio.file.Paths;
 import java.util.UUID;
 
-import org.apache.commons.io.FileUtils;
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.filecache.DistributedCache;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.hotels.bdp.circustrain.api.CircusTrainException;
+import com.hotels.bdp.circustrain.gcp.context.GCPSecurity;
 
 class GCPCredentialCopier {
   private static final Logger LOG = LoggerFactory.getLogger(GCPCredentialCopier.class);
+
   private static final String WORKING_DIRECTORY = System.getProperty("user.dir");
-  private static final String RANDOM_STRING = UUID.randomUUID().toString() + System.currentTimeMillis();
+  private static final String DEFAULT_HDFS_PREFIX = "hdfs:///tmp/ct-gcp-";
+  private static final String RANDOM_STRING = UUID.randomUUID().toString();
   private static final String CACHED_CREDENTIAL_NAME = "/ct-gcp-key-" + RANDOM_STRING + ".json";
-  private static final String HDFS_GS_CREDENTIAL_DIRECTORY = "hdfs:///tmp/ct-gcp-" + RANDOM_STRING;
-  private static final String HDFS_GS_CREDENTIAL_ABSOLUTE_PATH = HDFS_GS_CREDENTIAL_DIRECTORY + CACHED_CREDENTIAL_NAME;
 
   private final FileSystem fs;
   private final Configuration conf;
-  private final String credentialProvider;
+  private final GCPSecurity security;
+  private final String credentialsFileRelativePath;
+  private final String hdfsGsCredentialDirectory;
+  private final String hdfsGsCredentialAbsolutePath;
 
-  GCPCredentialCopier(FileSystem fs, Configuration conf, String credentialProvider) {
+  GCPCredentialCopier(FileSystem fs, Configuration conf, GCPSecurity security) {
     this.fs = fs;
     this.conf = conf;
-    if (credentialProvider == null) {
-      throw new IllegalArgumentException("credentialProvider must be set");
+    if (security == null || isBlank(security.getCredentialProvider())) {
+      throw new IllegalArgumentException("gcp-security credential-provider must be set");
+    }
+    this.security = security;
+    credentialsFileRelativePath = getCredentialProviderRelativePath();
+    hdfsGsCredentialDirectory = getHdfsGsCredentialDirectory();
+    hdfsGsCredentialAbsolutePath = getHdfsGsCredentialAbsolutePath();
+    LOG.debug("Credential Provider URI = {}", credentialsFileRelativePath);
+    LOG.debug("Temporary HDFS Google Cloud credential location set to {}", hdfsGsCredentialDirectory);
+    LOG.debug("HDFS Google Cloud credential path will be {}", hdfsGsCredentialAbsolutePath);
+  }
+
+  private String getHdfsGsCredentialDirectory() {
+    if (isBlank(security.getDistributedFileSystemWorkingDirectory())) {
+      return DEFAULT_HDFS_PREFIX + RANDOM_STRING;
+    } else {
+      return security.getDistributedFileSystemWorkingDirectory();
     }
-    this.credentialProvider = credentialProvider;
+  }
+
+  private String getCredentialProviderRelativePath() {
```

Comment:
You can add 
```if (new File(security.getCredentialProvider).isAbsolute()) {
           //relativize
     } else {
           //return unmodified path
     }
```
So that users can configure it either way rather than job failing if they provide an absolute path in the config

Reply:
That's a great idea, I'll add this now.